.circleci/config.yml

```diff
@@ -93,4 +95,32 @@ jobs:
             if [ $? -ne 0 ]; then
               echo "Unformatted code detected. Please run black --fast . locally and commit to format your code." >&2
               exit 1
-            fi
\ No newline at end of file
+            fi
+
+  ## ------------------ Mypy for Static Type Checking python code ------------------
+  mypy-type-checks:
+    parameters:
+      with_merge: # Part of the PR-merge workflow
+        type: boolean
+        default: false
+    docker:
+      # https://circleci.com/developer/images/image/cimg/python
+      - image: cimg/python:3.11.0
+    steps:
+      - when:
+          condition: << parameters.with_merge >>
+          steps:
+            - disable_build_if_not_pr
+      - checkout
+      # Run Pyre type checking
+      - run:
+          name: Run Static type checking
+          command: |
+            set -x
+            pip wheel --wheel-dir=./wheel-dir -r requirements-type-checks.txt
+            pip install --find-links=./wheel-dir -r requirements-type-checks.txt
+            mypy $(git diff --name-only --diff-filter=A origin/master HEAD | grep '\.py$') && mypy
+            if [ $? -ne 0 ]; then
+              echo " MyPy finds static type error(s), in the Python codebase. Please fix all type-related erros before commiting the code." >&2
+              exit 1
+            fi
```

Comment:
It looks like there is an xunit file which can be uploaded to test results as well:
```
--junit-xml JUNIT_XML
Causes mypy to generate a JUnit XML test result document with type checking results. This can make it easier to integrate mypy with continuous integration (CI) tools.
```
The errors in that report can used to fail the build.  Though I think the report could use some work.

Reply:
Added this option in the global section ([mypy]) of 'mypy.ini'